# Differences here are expected due to the "Enhanced Hints" feature
--- a/expected/updatable_views.out
+++ b/expected/updatable_views.out
@@ -1425,10 +1425,12 @@
 
 INSERT INTO base_tbl VALUES (3, 'Row 3', 3.0); -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT INSERT ON public.base_tbl TO regress_view_user2;
 INSERT INTO rw_view1 VALUES ('Row 3', 3.0, 3); -- not allowed
 ERROR:  permission denied for view rw_view1
 INSERT INTO rw_view2 VALUES ('Row 3', 3.0, 3); -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT INSERT ON public.base_tbl TO regress_view_user2;
 MERGE INTO rw_view1 t
   USING (VALUES ('Row 3', 3.0, 3)) AS v(b,c,a) ON t.aa = v.a
   WHEN NOT MATCHED THEN INSERT VALUES (v.b, v.c, v.a); -- not allowed
@@ -1437,15 +1439,18 @@
   USING (VALUES ('Row 3', 3.0, 3)) AS v(b,c,a) ON t.aa = v.a
   WHEN NOT MATCHED THEN INSERT VALUES (v.b, v.c, v.a); -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT INSERT ON public.base_tbl TO regress_view_user2;
 UPDATE base_tbl SET a=a, c=c; -- ok
 UPDATE base_tbl SET b=b; -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT UPDATE ON public.base_tbl TO regress_view_user2;
 UPDATE rw_view1 SET bb=bb, cc=cc; -- ok
 UPDATE rw_view1 SET aa=aa; -- not allowed
 ERROR:  permission denied for view rw_view1
 UPDATE rw_view2 SET aa=aa, cc=cc; -- ok
 UPDATE rw_view2 SET bb=bb; -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT UPDATE ON public.base_tbl TO regress_view_user2;
 MERGE INTO rw_view1 t USING (VALUES (1)) AS v(a) ON t.aa = v.a
   WHEN MATCHED THEN UPDATE SET bb = bb, cc = cc; -- ok
 MERGE INTO rw_view1 t USING (VALUES (1)) AS v(a) ON t.aa = v.a
@@ -1456,12 +1461,15 @@
 MERGE INTO rw_view2 t USING (VALUES (1)) AS v(a) ON t.aa = v.a
   WHEN MATCHED THEN UPDATE SET bb = bb; -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT UPDATE ON public.base_tbl TO regress_view_user2;
 DELETE FROM base_tbl; -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT DELETE ON public.base_tbl TO regress_view_user2;
 DELETE FROM rw_view1; -- not allowed
 ERROR:  permission denied for view rw_view1
 DELETE FROM rw_view2; -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT DELETE ON public.base_tbl TO regress_view_user2;
 RESET SESSION AUTHORIZATION;
 SET SESSION AUTHORIZATION regress_view_user1;
 GRANT INSERT, DELETE ON base_tbl TO regress_view_user2;
@@ -1495,19 +1503,24 @@
 SET SESSION AUTHORIZATION regress_view_user2;
 INSERT INTO base_tbl VALUES (5, 'Row 5', 5.0); -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT INSERT ON public.base_tbl TO regress_view_user2;
 INSERT INTO rw_view1 VALUES ('Row 5', 5.0, 5); -- ok
 INSERT INTO rw_view2 VALUES ('Row 6', 6.0, 6); -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT INSERT ON public.base_tbl TO regress_view_user2;
 DELETE FROM base_tbl WHERE a=3; -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT DELETE ON public.base_tbl TO regress_view_user2;
 DELETE FROM rw_view1 WHERE aa=3; -- ok
 DELETE FROM rw_view2 WHERE aa=4; -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT DELETE ON public.base_tbl TO regress_view_user2;
 MERGE INTO rw_view1 t USING (VALUES (1)) AS v(a) ON t.aa = v.a
   WHEN MATCHED AND bb = 'xxx' THEN DELETE; -- ok
 MERGE INTO rw_view2 t USING (VALUES (1)) AS v(a) ON t.aa = v.a
   WHEN MATCHED AND bb = 'xxx' THEN DELETE; -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT DELETE ON public.base_tbl TO regress_view_user2;
 SELECT * FROM base_tbl;
  a |   b   | c 
 ---+-------+---
@@ -1527,13 +1540,17 @@
 CREATE VIEW rw_view1 AS SELECT * FROM base_tbl;
 SELECT * FROM rw_view1;  -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT SELECT ON public.base_tbl TO regress_view_user1;
 SELECT * FROM rw_view1 FOR UPDATE;  -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT SELECT, UPDATE ON public.base_tbl TO regress_view_user1;
 UPDATE rw_view1 SET b = 'foo' WHERE a = 1;  -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT SELECT, UPDATE ON public.base_tbl TO regress_view_user1;
 MERGE INTO rw_view1 t USING (VALUES (1)) AS v(a) ON t.a = v.a
   WHEN MATCHED THEN UPDATE SET b = 'foo'; -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT SELECT, UPDATE ON public.base_tbl TO regress_view_user1;
 SET SESSION AUTHORIZATION regress_view_user2;
 CREATE VIEW rw_view2 AS SELECT * FROM rw_view1;
 SELECT * FROM rw_view2;  -- not allowed
@@ -1556,11 +1573,14 @@
 
 SELECT * FROM rw_view1 FOR UPDATE;  -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT UPDATE ON public.base_tbl TO regress_view_user1;
 UPDATE rw_view1 SET b = 'foo' WHERE a = 1;  -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT UPDATE ON public.base_tbl TO regress_view_user1;
 MERGE INTO rw_view1 t USING (VALUES (1)) AS v(a) ON t.a = v.a
   WHEN MATCHED THEN UPDATE SET b = 'foo'; -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT UPDATE ON public.base_tbl TO regress_view_user1;
 SET SESSION AUTHORIZATION regress_view_user2;
 SELECT * FROM rw_view2;  -- not allowed
 ERROR:  permission denied for view rw_view1
@@ -1648,11 +1668,14 @@
 
 SELECT * FROM rw_view1 FOR UPDATE;  -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT UPDATE ON public.base_tbl TO regress_view_user1;
 UPDATE rw_view1 SET b = 'foo' WHERE a = 1;  -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT UPDATE ON public.base_tbl TO regress_view_user1;
 MERGE INTO rw_view1 t USING (VALUES (1)) AS v(a) ON t.a = v.a
   WHEN MATCHED THEN UPDATE SET b = 'foo'; -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT UPDATE ON public.base_tbl TO regress_view_user1;
 SET SESSION AUTHORIZATION regress_view_user2;
 SELECT * FROM rw_view2;
  a |  b  | c 
@@ -1662,11 +1685,14 @@
 
 SELECT * FROM rw_view2 FOR UPDATE;  -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT UPDATE ON public.base_tbl TO regress_view_user2;
 UPDATE rw_view2 SET b = 'bar' WHERE a = 1;  -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT UPDATE ON public.base_tbl TO regress_view_user2;
 MERGE INTO rw_view2 t USING (VALUES (1)) AS v(a) ON t.a = v.a
   WHEN MATCHED THEN UPDATE SET b = 'foo'; -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT UPDATE ON public.base_tbl TO regress_view_user2;
 RESET SESSION AUTHORIZATION;
 DROP TABLE base_tbl CASCADE;
 NOTICE:  drop cascades to 2 other objects
@@ -1684,21 +1710,28 @@
 SET SESSION AUTHORIZATION regress_view_user2;
 SELECT * FROM base_tbl; -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT SELECT ON public.base_tbl TO regress_view_user2;
 SELECT * FROM rw_view1; -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT SELECT ON public.base_tbl TO regress_view_user2;
 INSERT INTO base_tbl VALUES (3, 'Row 3', 3.0); -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT INSERT ON public.base_tbl TO regress_view_user2;
 INSERT INTO rw_view1 VALUES ('Row 3', 3.0, 3); -- not allowed
 ERROR:  permission denied for view rw_view1
 UPDATE base_tbl SET a=a; -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT SELECT, UPDATE ON public.base_tbl TO regress_view_user2;
 UPDATE rw_view1 SET bb=bb, cc=cc; -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT SELECT, UPDATE ON public.base_tbl TO regress_view_user2;
 MERGE INTO rw_view1 t USING (VALUES (1)) AS v(a) ON t.aa = v.a
   WHEN MATCHED THEN UPDATE SET bb = bb; -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT SELECT, UPDATE ON public.base_tbl TO regress_view_user2;
 DELETE FROM base_tbl; -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT DELETE ON public.base_tbl TO regress_view_user2;
 DELETE FROM rw_view1; -- not allowed
 ERROR:  permission denied for view rw_view1
 MERGE INTO rw_view1 t USING (VALUES (1)) AS v(a) ON t.aa = v.a
@@ -1725,6 +1758,7 @@
 UPDATE base_tbl SET a=a, c=c; -- ok
 UPDATE base_tbl SET b=b; -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT UPDATE ON public.base_tbl TO regress_view_user2;
 UPDATE rw_view1 SET cc=cc; -- ok
 MERGE INTO rw_view1 t USING (VALUES (1)) AS v(a) ON t.aa = v.a
   WHEN MATCHED THEN UPDATE SET cc = cc; -- ok
@@ -1732,12 +1766,14 @@
 ERROR:  permission denied for view rw_view1
 UPDATE rw_view1 SET bb=bb; -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT UPDATE ON public.base_tbl TO regress_view_user2;
 MERGE INTO rw_view1 t USING (VALUES (1)) AS v(a) ON t.aa = v.a
   WHEN MATCHED THEN UPDATE SET aa = aa; -- not allowed
 ERROR:  permission denied for view rw_view1
 MERGE INTO rw_view1 t USING (VALUES (1)) AS v(a) ON t.aa = v.a
   WHEN MATCHED THEN UPDATE SET bb = bb; -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT UPDATE ON public.base_tbl TO regress_view_user2;
 SET SESSION AUTHORIZATION regress_view_user1;
 GRANT INSERT, DELETE ON base_tbl TO regress_view_user2;
 SET SESSION AUTHORIZATION regress_view_user2;
@@ -1756,11 +1792,14 @@
 SET SESSION AUTHORIZATION regress_view_user2;
 INSERT INTO rw_view1 VALUES ('Row 4', 4.0, 4); -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT INSERT ON public.base_tbl TO regress_view_user2;
 DELETE FROM rw_view1 WHERE aa=2; -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT DELETE ON public.base_tbl TO regress_view_user2;
 MERGE INTO rw_view1 t USING (VALUES (1)) AS v(a) ON t.aa = v.a
   WHEN MATCHED THEN DELETE; -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT DELETE ON public.base_tbl TO regress_view_user2;
 SET SESSION AUTHORIZATION regress_view_user1;
 GRANT INSERT, DELETE ON base_tbl TO regress_view_user2;
 SET SESSION AUTHORIZATION regress_view_user2;
@@ -1785,11 +1824,14 @@
 ALTER VIEW rw_view1 SET (security_invoker = true);
 SELECT * FROM rw_view1;  -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT SELECT ON public.base_tbl TO regress_view_user1;
 UPDATE rw_view1 SET aa=aa;  -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT SELECT, UPDATE ON public.base_tbl TO regress_view_user1;
 MERGE INTO rw_view1 t USING (VALUES (2, 'Row 2', 2.0)) AS v(a,b,c) ON t.aa = v.a
   WHEN NOT MATCHED THEN INSERT VALUES (v.b, v.c, v.a); -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT SELECT, INSERT ON public.base_tbl TO regress_view_user1;
 SET SESSION AUTHORIZATION regress_view_user2;
 CREATE VIEW rw_view2 AS SELECT cc AS ccc, aa AS aaa, bb AS bbb FROM rw_view1;
 GRANT SELECT, UPDATE ON rw_view2 TO regress_view_user3;
@@ -1813,11 +1855,13 @@
 UPDATE rw_view1 SET aa=aa, bb=bb;  -- ok
 UPDATE rw_view1 SET cc=cc;  -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT UPDATE ON public.base_tbl TO regress_view_user1;
 MERGE INTO rw_view1 t USING (VALUES (1)) AS v(a) ON t.aa = v.a
   WHEN MATCHED THEN UPDATE SET aa = aa, bb = bb; -- ok
 MERGE INTO rw_view1 t USING (VALUES (1)) AS v(a) ON t.aa = v.a
   WHEN MATCHED THEN UPDATE SET cc = cc; -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT UPDATE ON public.base_tbl TO regress_view_user1;
 SET SESSION AUTHORIZATION regress_view_user2;
 SELECT * FROM rw_view2;  -- not allowed
 ERROR:  permission denied for view rw_view1
@@ -1840,19 +1884,25 @@
 SET SESSION AUTHORIZATION regress_view_user2;
 SELECT * FROM rw_view2;  -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT SELECT ON public.base_tbl TO regress_view_user2;
 UPDATE rw_view2 SET bbb=bbb;  -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT SELECT, UPDATE ON public.base_tbl TO regress_view_user2;
 MERGE INTO rw_view2 t USING (VALUES (1)) AS v(a) ON t.aaa = v.a
   WHEN MATCHED THEN UPDATE SET bbb = bbb; -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT SELECT, UPDATE ON public.base_tbl TO regress_view_user2;
 SET SESSION AUTHORIZATION regress_view_user3;
 SELECT * FROM rw_view2;  -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT SELECT ON public.base_tbl TO regress_view_user3;
 UPDATE rw_view2 SET bbb=bbb;  -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT SELECT, UPDATE ON public.base_tbl TO regress_view_user3;
 MERGE INTO rw_view2 t USING (VALUES (1)) AS v(a) ON t.aaa = v.a
   WHEN MATCHED THEN UPDATE SET bbb = bbb; -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT SELECT, UPDATE ON public.base_tbl TO regress_view_user3;
 RESET SESSION AUTHORIZATION;
 GRANT SELECT ON base_tbl TO regress_view_user2;
 GRANT UPDATE (a, c) ON base_tbl TO regress_view_user2;
@@ -1867,6 +1917,7 @@
 ERROR:  permission denied for view rw_view1
 UPDATE rw_view2 SET bbb=bbb;  -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT UPDATE ON public.base_tbl TO regress_view_user2;
 UPDATE rw_view2 SET ccc=ccc;  -- ok
 MERGE INTO rw_view2 t USING (VALUES (1)) AS v(a) ON t.aaa = v.a
   WHEN MATCHED THEN UPDATE SET aaa = aaa; -- not allowed
@@ -1874,26 +1925,32 @@
 MERGE INTO rw_view2 t USING (VALUES (1)) AS v(a) ON t.aaa = v.a
   WHEN MATCHED THEN UPDATE SET bbb = bbb; -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT UPDATE ON public.base_tbl TO regress_view_user2;
 MERGE INTO rw_view2 t USING (VALUES (1)) AS v(a) ON t.aaa = v.a
   WHEN MATCHED THEN UPDATE SET ccc = ccc; -- ok
 SET SESSION AUTHORIZATION regress_view_user3;
 SELECT * FROM rw_view2;  -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT SELECT ON public.base_tbl TO regress_view_user3;
 UPDATE rw_view2 SET aaa=aaa;  -- not allowed
 ERROR:  permission denied for view rw_view1
 UPDATE rw_view2 SET bbb=bbb;  -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT SELECT, UPDATE ON public.base_tbl TO regress_view_user3;
 UPDATE rw_view2 SET ccc=ccc;  -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT SELECT, UPDATE ON public.base_tbl TO regress_view_user3;
 MERGE INTO rw_view2 t USING (VALUES (1)) AS v(a) ON t.aaa = v.a
   WHEN MATCHED THEN UPDATE SET aaa = aaa; -- not allowed
 ERROR:  permission denied for view rw_view1
 MERGE INTO rw_view2 t USING (VALUES (1)) AS v(a) ON t.aaa = v.a
   WHEN MATCHED THEN UPDATE SET bbb = bbb; -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT SELECT, UPDATE ON public.base_tbl TO regress_view_user3;
 MERGE INTO rw_view2 t USING (VALUES (1)) AS v(a) ON t.aaa = v.a
   WHEN MATCHED THEN UPDATE SET ccc = ccc; -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT SELECT, UPDATE ON public.base_tbl TO regress_view_user3;
 RESET SESSION AUTHORIZATION;
 GRANT SELECT ON base_tbl TO regress_view_user3;
 GRANT UPDATE (a, c) ON base_tbl TO regress_view_user3;
@@ -1908,6 +1965,7 @@
 ERROR:  permission denied for view rw_view1
 UPDATE rw_view2 SET bbb=bbb;  -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT UPDATE ON public.base_tbl TO regress_view_user3;
 UPDATE rw_view2 SET ccc=ccc;  -- ok
 MERGE INTO rw_view2 t USING (VALUES (1)) AS v(a) ON t.aaa = v.a
   WHEN MATCHED THEN UPDATE SET aaa = aaa; -- not allowed
@@ -1915,6 +1973,7 @@
 MERGE INTO rw_view2 t USING (VALUES (1)) AS v(a) ON t.aaa = v.a
   WHEN MATCHED THEN UPDATE SET bbb = bbb; -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT UPDATE ON public.base_tbl TO regress_view_user3;
 MERGE INTO rw_view2 t USING (VALUES (1)) AS v(a) ON t.aaa = v.a
   WHEN MATCHED THEN UPDATE SET ccc = ccc; -- ok
 RESET SESSION AUTHORIZATION;
@@ -1922,11 +1981,14 @@
 SET SESSION AUTHORIZATION regress_view_user1;
 SELECT * FROM rw_view1;  -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT SELECT ON public.base_tbl TO regress_view_user1;
 UPDATE rw_view1 SET aa=aa;  -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT SELECT, UPDATE ON public.base_tbl TO regress_view_user1;
 MERGE INTO rw_view1 t USING (VALUES (1)) AS v(a) ON t.aa = v.a
   WHEN MATCHED THEN UPDATE SET aa = aa; -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT SELECT, UPDATE ON public.base_tbl TO regress_view_user1;
 SET SESSION AUTHORIZATION regress_view_user2;
 SELECT * FROM rw_view2;  -- ok
  ccc | aaa |  bbb  
@@ -1938,6 +2000,7 @@
 ERROR:  permission denied for view rw_view1
 UPDATE rw_view2 SET bbb=bbb;  -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT UPDATE ON public.base_tbl TO regress_view_user2;
 UPDATE rw_view2 SET ccc=ccc;  -- ok
 MERGE INTO rw_view2 t USING (VALUES (1)) AS v(a) ON t.aaa = v.a
   WHEN MATCHED THEN UPDATE SET aaa = aaa; -- not allowed
@@ -1945,6 +2008,7 @@
 MERGE INTO rw_view2 t USING (VALUES (1)) AS v(a) ON t.aaa = v.a
   WHEN MATCHED THEN UPDATE SET bbb = bbb; -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT UPDATE ON public.base_tbl TO regress_view_user2;
 MERGE INTO rw_view2 t USING (VALUES (1)) AS v(a) ON t.aaa = v.a
   WHEN MATCHED THEN UPDATE SET ccc = ccc; -- ok
 SET SESSION AUTHORIZATION regress_view_user3;
@@ -1958,6 +2022,7 @@
 ERROR:  permission denied for view rw_view1
 UPDATE rw_view2 SET bbb=bbb;  -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT UPDATE ON public.base_tbl TO regress_view_user3;
 UPDATE rw_view2 SET ccc=ccc;  -- ok
 MERGE INTO rw_view2 t USING (VALUES (1)) AS v(a) ON t.aaa = v.a
   WHEN MATCHED THEN UPDATE SET aaa = aaa; -- not allowed
@@ -1965,6 +2030,7 @@
 MERGE INTO rw_view2 t USING (VALUES (1)) AS v(a) ON t.aaa = v.a
   WHEN MATCHED THEN UPDATE SET bbb = bbb; -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT UPDATE ON public.base_tbl TO regress_view_user3;
 MERGE INTO rw_view2 t USING (VALUES (1)) AS v(a) ON t.aaa = v.a
   WHEN MATCHED THEN UPDATE SET ccc = ccc; -- ok
 RESET SESSION AUTHORIZATION;
@@ -1972,21 +2038,26 @@
 SET SESSION AUTHORIZATION regress_view_user2;
 SELECT * FROM rw_view2;  -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT SELECT ON public.base_tbl TO regress_view_user2;
 UPDATE rw_view2 SET aaa=aaa;  -- not allowed
 ERROR:  permission denied for view rw_view1
 UPDATE rw_view2 SET bbb=bbb;  -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT SELECT, UPDATE ON public.base_tbl TO regress_view_user2;
 UPDATE rw_view2 SET ccc=ccc;  -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT SELECT, UPDATE ON public.base_tbl TO regress_view_user2;
 MERGE INTO rw_view2 t USING (VALUES (1)) AS v(a) ON t.aaa = v.a
   WHEN MATCHED THEN UPDATE SET aaa = aaa; -- not allowed
 ERROR:  permission denied for view rw_view1
 MERGE INTO rw_view2 t USING (VALUES (1)) AS v(a) ON t.aaa = v.a
   WHEN MATCHED THEN UPDATE SET bbb = bbb; -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT SELECT, UPDATE ON public.base_tbl TO regress_view_user2;
 MERGE INTO rw_view2 t USING (VALUES (1)) AS v(a) ON t.aaa = v.a
   WHEN MATCHED THEN UPDATE SET ccc = ccc; -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT SELECT, UPDATE ON public.base_tbl TO regress_view_user2;
 SET SESSION AUTHORIZATION regress_view_user3;
 SELECT * FROM rw_view2;  -- ok
  ccc | aaa |  bbb  
@@ -1998,6 +2069,7 @@
 ERROR:  permission denied for view rw_view1
 UPDATE rw_view2 SET bbb=bbb;  -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT UPDATE ON public.base_tbl TO regress_view_user3;
 UPDATE rw_view2 SET ccc=ccc;  -- ok
 MERGE INTO rw_view2 t USING (VALUES (1)) AS v(a) ON t.aaa = v.a
   WHEN MATCHED THEN UPDATE SET aaa = aaa; -- not allowed
@@ -2005,6 +2077,7 @@
 MERGE INTO rw_view2 t USING (VALUES (1)) AS v(a) ON t.aaa = v.a
   WHEN MATCHED THEN UPDATE SET bbb = bbb; -- not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT UPDATE ON public.base_tbl TO regress_view_user3;
 MERGE INTO rw_view2 t USING (VALUES (1)) AS v(a) ON t.aaa = v.a
   WHEN MATCHED THEN UPDATE SET ccc = ccc; -- ok
 RESET SESSION AUTHORIZATION;
@@ -3882,6 +3955,7 @@
 insert into rw_view1 values ('yyy',2.0,1)
   on conflict (aa) do update set bb = excluded.cc; -- Not allowed
 ERROR:  permission denied for view rw_view1
+HINT:  Grant the required privileges to the current role with: GRANT SELECT ON public.rw_view1 TO regress_view_user2;
 insert into rw_view1 values ('yyy',2.0,1)
   on conflict (aa) do update set bb = rw_view1.cc; -- Not allowed
 ERROR:  permission denied for view rw_view1
@@ -3918,9 +3992,11 @@
 insert into rw_view2 (aa,bb) values (1,'xxx')
   on conflict (aa) do update set bb = excluded.bb; -- Not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT SELECT ON public.base_tbl TO regress_view_user2;
 insert into rw_view2 (aa,bb) values (1,'xxx')
   on conflict (aa) do select returning 1;          -- Not allowed
 ERROR:  permission denied for table base_tbl
+HINT:  Grant the required privileges to the current role with: GRANT SELECT ON public.base_tbl TO regress_view_user2;
 create view rw_view3 as select b as bb, a as aa from base_tbl;
 insert into rw_view3 (aa,bb) values (1,'xxx')
   on conflict (aa) do update set bb = excluded.bb; -- OK
